pkg/archive/changes_other.go

```diff
@@ -86,8 +91,12 @@ func collectFileInfo(sourceDir string, idMappings *idtools.IDMappings) (*FileInf
 		if err != nil {
 			return err
 		}
-		info.stat = s
 
+		if s.Dev() != sourceStat.Dev() {
```

Comment:
Is this also applicable for `*nix` ? or does this additional check `if os == freebsd`

Reply:
The st_dev field of struct stat is common across all *nix since at least v7. I added the field to StatT for *unix but so far I only initialised it on FreeBSD and Linux - I can go back and add it for darwin, openbsd and solaris but have no good way of testing.

The additional check is currently only needed for FreeBSD - on Linux, the use of mount namespaces seems to avoid the problem of traversing into sub-mounts.